src/core/project/qgsproject.cpp

```diff
@@ -685,11 +685,14 @@ void QgsProject::registerTranslatableObjects( QgsTranslationContext *translation
         }
         if ( field.editorWidgetSetup().type() == QStringLiteral( "ValueMap" ) )
         {
-          const QList<QVariant> valueList = field.editorWidgetSetup().config().value( QStringLiteral( "map" ) ).toList();
-
-          for ( int i = 0, row = 0; i < valueList.count(); i++, row++ )
+          if ( field.editorWidgetSetup().config().value( QStringLiteral( "map" ) ).canConvert<QList<QVariant>>() )
```

Comment:
Beside using QList<QVariant> consequently I added this line. Otherwise it would just had returned an empty list, but with this it's cleaner...